enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+
+    def my_each
+        for i in 0..self.length-1
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0..self.length-1
+            yield(self[i],i)
+        end
+    end
+
+    def my_select
+        arr = []
+        for i in 1..self.length
+            if yield(i)
+                arr.push(i)
+            end
+        end
+        arr
+    end
+
+    def my_all?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_any?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_none?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = false
+            end
+        end
+        results
+    end
```

Comment:
* [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
  ```ruby
    array = [1,2,5,4,7]
    p array.my_none? == array.none? # false
  ```
* [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_none?(Integer) == array.none?(Integer) # false
  ```
* [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(/c/) == array.none?(/c/) # false
  ```
* [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(5) == array.none?(5) # false
  ```